modules/GenericPowermeter/serialization.hpp

```diff
@@ -0,0 +1,178 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Qwello GmbH and Contributors to EVerest
+
+/// @brief Serialization and deserialization for the modbus interface.
+///
+/// The format has a major quirk: Even though our exchanged data is four bytes
+/// wide, the modbus implementation only uses the first two bytes, effectively
+/// casting every element in the exchanged data from uint32_t to uint16_t. If
+/// this ever gets changed, the code below must be equally adjusted.
+
+#include <chrono>
+#include <everest/logging.hpp>
+#include <optional>
+#include <stdexcept>
+#include <string>
+#include <type_traits>
+#include <vector>
+// TODO(ddo) We don't need the utils types but otherwise our compilation fails
+// since the generated files aren't self contained (they need json to be
+// externally included)...
+#include <utils/types.hpp>
+
+#include <generated/types/serial_comm_hub_requests.hpp>
+
+namespace internal {
+
+using Result = types::serial_comm_hub_requests::Result;
+
+inline void check_message(const Result& _message) {
+    if (!_message.value.has_value())
+        throw std::invalid_argument("The `value` must be set");
+}
+
+/// @brief Deserialization for small integral types (less or equal to two bytes).
+/// @param _message The message read.
+/// @param _result The result which will contain the deserialized data.
+template <typename Output, std::enable_if_t<std::is_integral<Output>::value && (sizeof(Output) <= 2), bool> = true>
+inline void deserialize(const Result& _message, Output& _result) {
```

Comment:
why are you using `_` at the beginning of your variables? For one, that is confusing to read, for two in c++:

> Each identifier that begins with an underscore is reserved to the implementation for use as a name in the global namespace.

From: https://eel.is/c++draft/lex.name#3.2.

So you should never start your identfiers with an underscore - unless you implement a c++ standard library or a c++ compiler. Same comment for all variables that start with `_` in your code here.